modules/tinymce/src/plugins/advlist/main/ts/api/Options.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import { Type } from '@ephox/katamari';
+
+import Editor from 'tinymce/core/api/Editor';
+import { EditorOptions } from 'tinymce/core/api/OptionTypes';
+
+const option: {
+  <K extends keyof EditorOptions>(name: K): (editor: Editor) => EditorOptions[K] | undefined;
+  <T>(name: string): (editor: Editor) => T | undefined;
+} = (name: string) => (editor: Editor) =>
+  editor.options.get(name);
+
+const register = (editor: Editor): void => {
+  const registerOption = editor.options.register;
+
+  const stringListProcessor = (value: unknown) => {
```

Comment:
I can already tell "composable built-in processors" are going to be high on my wishlist for this feature. Maybe not for 6.0, but at some point. Everything here except `value.split(/[ ,]/)` could come from the core.

Reply:
Yeah it would definitely be quite useful. There's quite a few places we just need to check if it's a `string` or `boolean` (or something similar to that).